test/parallel/test-fs-error-messages.js

```diff
@@ -9,202 +9,156 @@ const existingFile2 = path.join(common.fixturesDir, 'create-file.js');
+function testEnoentError(file, endMessage, syscal, err) {
+  const sufix = (endMessage) ? endMessage : '';
 
-fs.stat(fn, function(err) {
+  assert(err instanceof Error);
   assert.strictEqual(fn, err.path);
-  assert.ok(0 <= err.message.indexOf(fn));
-});
+  assert.strictEqual(
+    err.message,
+    `ENOENT: no such file or directory, ${syscal} '${file}'${sufix}`
+  );
 
-fs.lstat(fn, function(err) {
-  assert.ok(0 <= err.message.indexOf(fn));
-});
+  return true;
+}
 
-fs.readlink(fn, function(err) {
-  assert.ok(0 <= err.message.indexOf(fn));
-});
+// Test all operations failing with EEXIST errors
+function testEexistError(source, dest, syscal, err) {
+  const sufix = (dest) ? ` -> '${dest}'` : '';
 
-fs.link(fn, 'foo', function(err) {
-  assert.ok(0 <= err.message.indexOf(fn));
-});
+  assert(err instanceof Error);
+  assert.strictEqual(source, err.path);
+  assert.strictEqual(
+    err.message,
+    `EEXIST: file already exists, ${syscal} '${source}'${sufix}`
+  );
 
-fs.link(existingFile, existingFile2, function(err) {
-  assert.ok(0 <= err.message.indexOf(existingFile));
-  assert.ok(0 <= err.message.indexOf(existingFile2));
-});
+  return true;
+}
 
-fs.symlink(existingFile, existingFile2, function(err) {
-  assert.ok(0 <= err.message.indexOf(existingFile));
-  assert.ok(0 <= err.message.indexOf(existingFile2));
-});
+// Test all operations failing with ENOTEMPTY errors
+function testEnoemptyError(source, dest, err) {
+  assert(err instanceof Error);
+  assert.strictEqual(source, err.path);
+  assert.strictEqual(
+    err.message,
+    `ENOTEMPTY: directory not empty, rename '${source}' ` +
+    `-> '${dest}'`
+  );
+
+  return true;
+}
 
-fs.unlink(fn, function(err) {
-  assert.ok(0 <= err.message.indexOf(fn));
-});
+// Test all operations failing with ENOTDIR errors
+function testEnotdirError(dir, err) {
+  assert(err instanceof Error);
+  assert.strictEqual(dir, err.path);
+  assert.strictEqual(
+    err.message,
+    `ENOTDIR: not a directory, rmdir '${dir}'`
+  );
 
-fs.rename(fn, 'foo', function(err) {
-  assert.ok(0 <= err.message.indexOf(fn));
-});
+  return true;
+}
 
-fs.rename(existingDir, existingDir2, function(err) {
-  assert.ok(0 <= err.message.indexOf(existingDir));
-  assert.ok(0 <= err.message.indexOf(existingDir2));
-});
+// Generating ENOENTS errors
+fs.stat(fn, (err) => testEnoentError(fn, '', 'stat', err));
+fs.lstat(fn, (err) => testEnoentError(fn, '', 'lstat', err));
+fs.readlink(fn, (err) => testEnoentError(fn, '', 'readlink', err));
+fs.link(fn, 'foo', (err) => testEnoentError(fn, ' -> \'foo\'', 'link', err));
+fs.unlink(fn, (err) => testEnoentError(fn, '', 'unlink', err));
+fs.rmdir(fn, (err) => testEnoentError(fn, '', 'rmdir', err));
+fs.chmod(fn, 0o666, (err) => testEnoentError(fn, '', 'chmod', err));
+fs.open(fn, 'r', 0o666, (err) => testEnoentError(fn, '', 'open', err));
+fs.readFile(fn, (err) => testEnoentError(fn, '', 'open', err));
+fs.readdir(fn, (err) => testEnoentError(fn, '', 'scandir', err));
```

Comment:
Any reason why the `syscall` parameter (3rd) is passed as `scandir` and not `readdir` ?
Also the `readFile` is passing `open`